x/ecocredit/core/msg_bridge.go

```diff
@@ -0,0 +1,49 @@
+package core
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+var _ legacytx.LegacyMsg = &MsgBridge{}
+
+// Route implements the LegacyMsg interface.
+func (m MsgBridge) Route() string { return sdk.MsgTypeURL(&m) }
+
+// Type implements the LegacyMsg interface.
+func (m MsgBridge) Type() string { return sdk.MsgTypeURL(&m) }
+
+// GetSignBytes implements the LegacyMsg interface.
+func (m MsgBridge) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ecocredit.ModuleCdc.MustMarshalJSON(&m))
+}
+
+// ValidateBasic does a sanity check on the provided data.
+func (m *MsgBridge) ValidateBasic() error {
+
+	if err := m.MsgCancel.ValidateBasic(); err != nil {
+		return err
+	}
+
+	if len(m.BridgeContract) == 0 {
+		return sdkerrors.ErrInvalidRequest.Wrap("bridge contract should not be empty")
+	}
+
+	if len(m.BridgeRecipient) == 0 {
```

Comment:
Do we need additional validations for these fields?

Reply:
Good question. We are only planning to support bridging to Polygon in the short-term but we are planning to bridge to other chains in the future. I think we should validate for an Ethereum address.

Updating the validation check could be done in a patch release if we need to change it but better to be more cautious given we are cancelling credits and this action cannot be undone.

The following is an example from the [Go Ethereum Documentation](https://goethereumbook.org/en/address-check/):

```
re := regexp.MustCompile("^0x[0-9a-fA-F]{40}$")

fmt.Printf("is valid: %v\n", re.MatchString("0x323b5d4c32345ced77393b3530b1eed0f346429d")) // is valid: true
fmt.Printf("is valid: %v\n", re.MatchString("0xZYXb5d4c32345ced77393b3530b1eed0f346429d")) // is valid: false
```